src/server/services/crud.ts

```diff
@@ -0,0 +1,60 @@
+import { Model, Document } from "mongoose";
+import { CrudTypeNotFound } from "../errors";
+import BaseModel from "../interfaces/base";
+
+
+export interface ICrudService<TDocument extends Document & TIInstance, TIInstance extends BaseModel> {
```

Comment:
@Evilart86 What is TIInstance? can we simplify this name ? Also prefix I

Reply:
Type of instance interface

Reply:
@Evilart86 remove prefix I